zcash_primitives/src/zip32/fingerprint.rs

```diff
@@ -0,0 +1,53 @@
+//! Seed Fingerprints according to ZIP 32
+//!
+//! Implements section `Seed Fingerprints` of Shielded Hierarchical Deterministic Wallets (ZIP 32)
+//!
+//! [Section Seed Fingerprints]: https://zips.z.cash/zip-0032#seed-fingerprints
+
+use blake2b_simd::Params as Blake2bParams;
+
+pub const ZIP32_SEED_FP_PERSONALIZATION: &[u8; 16] = b"Zcash_HD_Seed_FP";
+
+pub struct SeedFingerprint([u8; 32]);
+
+impl SeedFingerprint {
+    pub fn from_seed(seed_bytes: &[u8]) -> SeedFingerprint {
```

Comment:
```suggestion
    /// Return the seed fingerprint of the wallet as defined in
    /// <https://zips.z.cash/zip-0032#seed-fingerprints>.
    pub fn from_seed(seed_bytes: &[u8]) -> SeedFingerprint {
```

Reply:
Given the restriction on `seed_bytes.len()`, I think that this should return an `Option<SeedFingerprint>` such that it returns `None` if the length of the seed exceeds what can be represented in a lead byte.